fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/postgresql/PostgreSqlParameterNamesDAO.java

```diff
@@ -6,75 +6,63 @@
+    public PostgreSqlParameterNamesDAO(Connection c) {
         super(c);
-        this.fhirRefSequenceDAO = fsd;
-    }
-
-    @Override
-    public int readOrAddParameterNameId(String parameterName) throws FHIRPersistenceDataAccessException  {
-        // As the system is concurrent, we have to handle cases where another thread
-        // might create the entry after we selected and found nothing
-        Integer result = getParameterId(parameterName);
-
-        // Create the resource if we don't have it already (set by the continue handler)
-        if (result == null) {
-            try {
-                result = fhirRefSequenceDAO.nextValue();
-
-                final String INS = "INSERT INTO parameter_names (parameter_name_id, parameter_name) VALUES (?, ?)";
-                try (PreparedStatement stmt = getConnection().prepareStatement(INS)) {
-                    // bind parameters
-                    stmt.setInt(1, result);
-                    stmt.setString(2, parameterName);
-                    stmt.executeUpdate();
-                }
-            } catch (SQLException e) {
-                throw new FHIRPersistenceDataAccessException("Error while getting or inserting parameterName '" + parameterName + "'", e);
-            }
-        }
-
-        // cannot be null, so safe to return as an int
-        return result;
     }
 
     /**
-     * Read the id for the named type
+     * Calls a stored procedure to read the name contained in the passed Parameter in the Parameter_Names table.
+     * If it's not in the DB, it will be stored and a unique id will be returned.
      * @param parameterName
-     * @return the database id, or null if the named record is not found
+     * @return The generated id of the stored system.
      * @throws FHIRPersistenceDataAccessException
      */
-    protected Integer getParameterId(String parameterName) throws FHIRPersistenceDataAccessException {
-        Integer result;
+    @Override
+    public int readOrAddParameterNameId(String parameterName) throws FHIRPersistenceDataAccessException  {
+        final String METHODNAME = "readOrAddParameterNameId";
+        log.entering(CLASSNAME, METHODNAME);
 
-        String sql = "SELECT parameter_name_id FROM parameter_names WHERE parameter_name = ?";
+        int parameterNameId;
+        String currentSchema;
+        String stmtString;
+        String errMsg = "Failure storing search parameter name id: name=" + parameterName;
+        long dbCallStartTime;
+        double dbCallDuration;
 
-        try (PreparedStatement stmt = getConnection().prepareStatement(sql)) {
-            stmt.setString(1, parameterName);
-            ResultSet rs = stmt.executeQuery();
-            if (rs.next()) {
-                result = rs.getInt(1);
-            }  else {
-                result = null;
+        try {
+            // TODO: schema should be known by application. Fix to avoid an extra round-trip.
```

Comment:
It should be set by the connection.
We can enforce this as part of the creation of the datasource

Reply:
agree, we can improve this in future PR.